### PR TITLE
Implement a link helper for links that open in new tab

### PR DESCRIPTION
**Pull request by cpjmcquillan**
- Append "(opens in new tab)" to link text
- Add target="_blank" to link
- Add rel="noreferrer noopener" to link

## David sign off
- [x] Looks good!

### CODE REVIEW

**Review by alexbowen — APPROVED** [review]
lgtm

File: config/locales/forms.yml [hunk 1]
```diff
@@ -425,7 +425,7 @@ en:
           "false": Use another application process or form
           true_html: >-
             Apply using the Teaching Vacancies application form.
-            <a class="govuk-link" target="_blank" href="/pages/job-application-preview">Preview the application form (opens in a new tab)</a>
+            <a class="govuk-link" target="_blank" href="/pages/job-application-preview">Preview the application form (opens in new tab)</a>
```
[the remark this answers is not in the document]
Comment (cpjmcquillan) [reply config/locales/forms.yml:428]: Theres a trade off between having a simple template (form builder looks up the translations for all these sorts of things for us), and not having any html in the template. Not sure what side we want to fall in this instance?

Either way, I think it's outside the scope of my PR! 😉

[the remark this answers is not in the document]
Comment (alexbowen) [reply config/locales/forms.yml:428]: ok im not really to bothered!

**Review by cesidio — APPROVED** [review]
😢

File: app/helpers/link_helper.rb [hunk 2]
```diff
@@ -4,4 +4,8 @@ def active_link_class(link_path)
 
     "govuk-header__navigation-item"
   end
+
+  def open_in_new_tab_link_to(text, href, **kwargs)
+    govuk_link_to("#{text} (opens in new tab)", href, target: "_blank", rel: "noreferrer noopener", **kwargs)
```
Comment (david-mears-dfe) [inline app/helpers/link_helper.rb:9]: Edge case but what happens if kwargs includes some other `rel` attributes? Do they overwrite or merge with the noferrer/noopener?

**Review by david-mears-dfe — APPROVED** [review]

**Review by david-mears-dfe — COMMENTED** [review]
Also I think noreferrer noopener are only relevant for outgoing links. Can we somehow tell if the href is to another website when it's passed in?

**Comment by david-mears-dfe** [comment]
> Also I think noreferrer noopener are only relevant for outgoing links. Can we somehow tell if the href is to another website when it's passed in?

e.g. `if href.match("teaching-vacancies")`

**Comment by cpjmcquillan** [comment]
> > Also I think noreferrer noopener are only relevant for outgoing links. Can we somehow tell if the href is to another website when it's passed in?
> 
> e.g. `if href.match("teaching-vacancies")`

Perhaps we could and perhaps we should. I'm wary about a helper knowing or caring much about the domain though.

Could we pass in an `external: true` option? Although at that point maybe we're better off sticking with what we have now?

**Comment by david-mears-dfe** [comment]
Happy to merge and come back to it later

**Pull request merged**